File: patterns/lichtsteuerungssystem/rules-python/lichtsteuerungssystem_dynamisch.py
from datetime import datetime

from openhab import rule, Registry
from openhab.triggers import GenericCronTrigger


@rule(triggers=[GenericCronTrigger("0 0 18 ? * * *")])
class AdjustLightBrightnessBasedOnTimeOfDay:
    def execute(self, module, input):
        hour = datetime.now().hour
        if 18 <= hour < 22:
            Registry.getItem("Light1_Dimmer").sendCommand(80)  # Helligkeit auf 80%
        else:
            Registry.getItem("Light1_Dimmer").sendCommand(20)  # Helligkeit auf 20%

File: patterns/lichtsteuerungssystem/rules-js/lichtsteuerungssystem-dynamisch.js
rules.JSRule({
  name: "Adjust light brightness based on time of day",
  triggers: [triggers.GenericCronTrigger("0 0 18 ? * * *")],
  execute: (event) => {
    const hour = time.ZonedDateTime.now().hour();
    if (hour >= 18 && hour < 22) {
      items.getItem("Light1_Dimmer").sendCommand(80); // Helligkeit auf 80%
      console.log("Stunde=" + hour + " -> Light1_Dimmer 80%");
    } else {
      items.getItem("Light1_Dimmer").sendCommand(20); // Helligkeit auf 20%
      console.log("Stunde=" + hour + " -> Light1_Dimmer 20%");
    }
  }
});
